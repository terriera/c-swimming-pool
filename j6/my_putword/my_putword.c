/*
** my_putword.c for 2002 swimming pool
**
** Made by gumhead
**
*/

#include <unistd.h>

static void	my_putchar(char c)
{
  write(STDOUT_FILENO, &c, 1);
}

/*
** 1 << ((sizeof(int) * 8) - 1): sets the most significant bit and unset the
** others.
*/
void		my_putword(int n)
{
  unsigned int	mask;
  
  mask = 1 << ((sizeof(int) * 8) - 1);
  while (mask > 0)
  {
    my_putchar(n & mask ? '1' : '0');
    mask >>= 1;
  }
  my_putchar('\n');
}
