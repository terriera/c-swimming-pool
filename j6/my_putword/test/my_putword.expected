00000000000000000000000000000000
00000000000000000000000000000001
00000000000000000000000000000010
00000000000000000000000000000011
00000000000000000000000000000100
00000000000000000000000000000101
00000000000000000000000000000110
00000000000000000000000000000111
00000000000000000000000000001000
00000000000000000000000000001001
01111111111111111111111111111111
--------------------------------
00000000000000000000000000000000
11111111111111111111111111111111
11111111111111111111111111111110
11111111111111111111111111111101
11111111111111111111111111111100
11111111111111111111111111111011
11111111111111111111111111111010
11111111111111111111111111111001
11111111111111111111111111111000
11111111111111111111111111110111
10000000000000000000000000000000
